src/datachain/data_storage/sqlite.py

```diff
@@ -186,6 +186,12 @@ def _reconnect(self) -> None:
         self.db_file = db_file
         self.is_closed = False
 
+    def get_table(self, name: str) -> Table:
+        if self.is_closed:
+            # Reconnect in case of being closed previously.
+            self._reconnect()
```

Comment:
Ideally, this should be automatic. 

```
@property
def db(self):
  if self._db.is_closed:
      self._reconnect()
  return self._db
```

I see `executemany`, `execute_str` and other methods that are not reconnecting.
Although `engine` and `metadata` do complicate this.